esignet-core/src/test/java/io/mosip/esignet/core/ValidatorTest.java

```diff
@@ -647,4 +682,144 @@ public void test_ClientNameLangValidator_WithInValidDetail_thenFail(){
+
+		JsonNode addressNode = mapper.readValue(address, JsonNode.class);
+		JsonNode verifiedClaimNode = mapper.readValue(verifiedClaims, JsonNode.class);
+
+		Map<String, JsonNode> userinfoMap = new HashMap<>();
+		userinfoMap.put("address", addressNode);
+		userinfoMap.put("verified_claims", verifiedClaimNode);
+		Map<String, ClaimDetail> idTokenMap = new HashMap<>();
+
+		ClaimDetail claimDetail = new ClaimDetail("claim_value", null, true, "secondary");
+
+		idTokenMap.put("some_claim", claimDetail);
+
+		ClaimsV2 claimsV2 = new ClaimsV2();
+		claimsV2.setUserinfo(userinfoMap);
+		claimsV2.setId_token(idTokenMap);
+
+		Resource resource = Mockito.mock(Resource.class);
+		Mockito.when(resourceLoader.getResource(Mockito.anyString())).thenReturn(resource);
+		Mockito.when(resource.getInputStream()).thenReturn(new ByteArrayInputStream(claimSchema.getBytes()));
+
+		Assert.assertTrue(claimSchemaValidator.isValid(claimsV2, null));
+	}
+
+	@Test
+	public void test_ClaimSchemaValidator_withTrustFrameWorkAsNull_thenFail() throws IOException {
+
+		ObjectMapper mapper = new ObjectMapper();
+
+		ReflectionTestUtils.setField(claimSchemaValidator,"objectMapper",mapper);
+		ReflectionTestUtils.setField(claimSchemaValidator,"schemaUrl","http://localhost:8080/claims/schema");
+
+		String address="{\"essential\":true}";
+		String verifiedClaims="[{\"verification\":{\"trust_framework\":{\"value\":null}},\"claims\":{\"name\":null,\"email\":{\"essential\":true}}},{\"verification\":{\"trust_framework\":{\"value\":\"pwd\"}},\"claims\":{\"birthdate\":{\"essential\":true},\"address\":null}},{\"verification\":{\"trust_framework\":{\"value\":\"kaif\"}},\"claims\":{\"gender\":{\"essential\":true},\"email\":{\"essential\":true}}}]";
+
+		JsonNode addressNode = mapper.readValue(address, JsonNode.class);
+		JsonNode verifiedClaimNode = mapper.readValue(verifiedClaims, JsonNode.class);
+
+		Map<String, JsonNode> userinfoMap = new HashMap<>();
+		userinfoMap.put("address", addressNode);
+		userinfoMap.put("verified_claims", verifiedClaimNode);
+		Map<String, ClaimDetail> idTokenMap = new HashMap<>();
+
+
+		ClaimDetail claimDetail = new ClaimDetail("claim_value", null, true, "secondary");
+
+		idTokenMap.put("some_claim", claimDetail);
+		ClaimsV2 claimsV2 = new ClaimsV2();
+		claimsV2.setUserinfo(userinfoMap);
+		claimsV2.setId_token(idTokenMap);
+
+		Resource resource = Mockito.mock(Resource.class);
+		Mockito.when(resourceLoader.getResource(Mockito.anyString())).thenReturn(resource);
+		Mockito.when(resource.getInputStream()).thenReturn(new ByteArrayInputStream(claimSchema.getBytes()));
+
+		Assert.assertFalse(claimSchemaValidator.isValid(claimsV2, null));
+
+	}
+
+	@Test
+	public void test_ClaimSchemaValidator_withEssentialAsInteger_thenFail() throws IOException {
+
+		ObjectMapper mapper = new ObjectMapper();
```

Comment:
as this is used in more than one method better to declare objectmapper as class variable.